pkg/arvo/app/publish.hoon

```diff
@@ -373,842 +502,57 @@
-      ::
-          [%invites ~]
-        :_  this
-        :_  ~
-        :*  %pass  /invites  %agent  [our.bol %invite-store]  %watch
-            /invitatory/publish
-        ==
-      ==
-    ::
-        %fact
-      ?+  wir  (on-agent:def wir sin)
-          [%subscribe @ @ ~]
-        =/  who=@p     (slav %p i.t.wir)
-        =/  book-name  i.t.t.wir
-        ?>  ?=(%publish-notebook-delta p.cage.sin)
-        =^  cards  state
-          (handle-notebook-delta:main !<(notebook-delta q.cage.sin) state)
-        [cards this]
-      ::
-          [%groups ~]
-        =^  cards  state
-          (handle-group-update:main !<(update:group-store q.cage.sin))
-        [cards this]
-      ::
-          [%invites ~]
-        =^  cards  state
-          (handle-invite-update:main !<(update:inv q.cage.sin))
-        [cards this]
-      ::
-          [%collection *]
-        [~ this]
-      ==
-    ==
-  ::
+    ?.  ?=([%graph-migrate *] wire)
+      (on-agent:def wire sign)
+    =/  rid=resource
+      (de-path:resource t.wire)
+    ?.  ?=(%watch-ack -.sign)
+      (on-agent:def wire sign)
+    ?~  p.sign
+      ::  if watch acked successfully, then host has completed OTA, and
+      ::  we are safe to add it to the pull-hook
+      :_  this(migrate (~(del by migrate) rid))
+      ~[(poke-graph-pull:main %add entity.rid rid)]
+    ::  if nacked, then set a exponential backoff and retry
+    =/  nack-count=@ud
+      +((~(gut by migrate) rid 0))
+    ?:  (gte 24 nack-count)
+      ~&  >>>  "failed to migrate notebook {<rid>} to graph-store"
+      [~ this]
+    :_  this(migrate (~(put by migrate) rid nack-count))
+    ::  (bex 19) is roughly 6 days
+    =/  wakeup=@da
+      (add now.bol (bex (min 19 nack-count)))
+    [%pass wire %arvo %b %wait wakeup]~
+  ::   
   ++  on-arvo
-    |=  [wir=wire sin=sign-arvo]
+    |=  [=wire =sign-arvo]
     ^-  (quip card _this)
-    ?+  wir
-      (on-arvo:def wir sin)
-    ::
-        [%read %paths ~]
-      ?>  ?=([?(%b %c) %writ *] sin)
-      =/  rot=riot:clay  +>.sin
-      ?>  ?=(^ rot)
-      =^  cards  state
-        (read-paths:main u.rot)
-      [cards this]
-    ::
-        [%read %info *]
-      ?>  ?=([?(%b %c) %writ *] sin)
-      =/  rot=riot:clay  +>.sin
-      =^  cards  state
-        (read-info:main t.t.wir rot)
-      [cards this]
-    ::
-        [%read %note *]
-      ?>  ?=([?(%b %c) %writ *] sin)
-      =/  rot=riot:clay  +>.sin
-      =^  cards  state
-        (read-note:main t.t.wir rot)
-      [cards this]
-    ::
-        [%read %comment *]
-      ?>  ?=([?(%b %c) %writ *] sin)
-      =/  rot=riot:clay  +>.sin
-      =^  cards  state
-        (read-comment:main t.t.wir rot)
-      [cards this]
-    ::
-        [%bind ~]
-      [~ this]
-    ::
-        [%view-bind ~]
-      [~ this]
-    ==
+    ?.  ?=([%graph-migrate *] wire)
+      (on-arvo:def wire sign-arvo)
+    =/  rid=resource
+      (de-path:resource t.wire)
+    ?>  ?=([%b %wake *] sign-arvo)
+    ~?  ?=(^ error.sign-arvo)
```

Comment:
never seen this rune before???

Reply:
The elusive conditional sigpam

<img width="602" alt="image" src="https://user-images.githubusercontent.com/20846414/97499715-bcbe7100-1944-11eb-9a19-2cf95603153f.png">

Reply:
`=?` and `~?` come up surprisingly often